comp/core/tagger/types/entity_id.go

```diff
@@ -0,0 +1,148 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package types defines types used by the Tagger component.
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// EntityID represents a tagger entityID
+// An EntityID should be identified by a prefix and an id, and is represented as {prefix}://{id}
+type EntityID interface {
+	// GetID returns a prefix-specific id (i.e. an ID unique given prefix)
+	GetID() string
+	// GetPrefix returns the prefix of the EntityID
+	GetPrefix() EntityIDPrefix
+	// String returns a string representation of EntityID under the format {prefix}://{id}
+	String() string
+}
+
+// defaultEntityID implements EntityID as a plain string id
+type defaultEntityID string
+
+// GetID implements EntityID#GetID
+func (de defaultEntityID) GetID() string {
+	if strings.Contains(string(de), "://") {
+		return string(de)
+	}
+
+	parts := strings.Split(string(de), "://")
+	if len(parts) != 2 {
+		return ""
+	}
+
+	return parts[0]
+}
+
+// GetPrefix implements EntityID#GetPrefix
+func (de defaultEntityID) GetPrefix() EntityIDPrefix {
+	if strings.Contains(string(de), "://") {
```

Comment:
Should it be `if !strings.Contains(string(de), "://") {` ?

Reply:
Indeed. 

I think I should add unit tests for this interface to avoid such pitfalls.